src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx

```diff
@@ -60,17 +63,29 @@ export function OpenSavedQueryFlyout({
   const [selectedQuery, setSelectedQuery] = useState<SavedQuery | undefined>(undefined);
   const [searchQuery, setSearchQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
 
+  const currentTabIdRef = useRef(selectedTabId);
+
   const fetchAllSavedQueriesForSelectedTab = useCallback(async () => {
-    const allQueries = await savedQueryService.getAllSavedQueries();
-    const templateQueriesPresent = allQueries.some((q) => q.attributes.isTemplate);
-    const queriesForSelectedTab = allQueries.filter(
-      (q) =>
-        (selectedTabId === 'mutable-saved-queries' && !q.attributes.isTemplate) ||
-        (selectedTabId === 'template-saved-queries' && q.attributes.isTemplate)
-    );
-    setSavedQueries(queriesForSelectedTab);
-    setHasTemplateQueries(templateQueriesPresent);
-  }, [savedQueryService, selectedTabId, setSavedQueries]);
+    if (queryStringManager.getQuery()?.dataset?.type === 'SECURITY_LAKE') {
```

Comment:
Is Security Lake a specific aws type? Is there something more generic we could use or could we abstract this for any dataset type?

Reply:
We only want this flyout to be shown when the data source picker is a security lake connection

Reply:
@ashwin-pc @kavilla @AMoo-Miki Is this okay? Is there a better way of going about this?

Or can we track following up on implementing this in a generic way?

Reply:
Why are we hardcoding this? Isn't it just as easy to fetch it from the dataset config for the selected dataset?

Reply:
Updated in this commit: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8848/commits/e14f0be53fb68cb09743a01f2acd7ef730712929. So that we check `isTemplate` to determine if the tab should be shown.